coalib/core/Bear.py

```diff
@@ -130,6 +132,21 @@ class Bear:
     CAN_FIX = set()
     ASCIINEMA_URL = ''
     BEAR_DEPS = set()
+    """
+        Attributes for a new bear.
+
+        :param section:
+            The section object where bear settings are contained.
+        :param file_dict:
+            The file-dictionary containing a mapping of filenames to the
+            according file contents.
+        :raises RuntimeError:
+            Raised when bear requirements are not fulfilled.
+    """
+    section = attr.ib(hash=False)
+    file_dict = attr.ib(hash=False)
+    _dependency_results = attr.ib(init=False, default=defaultdict(list),
+                                  hash=False)
```

Comment:
@Makman2 attr is a decorator which generates an __init__ flexibly . The whole point is not only about not having to write an init. We can utilize the properties of attrs .Like , we can obtain all attributes as a dict or json .We can attach hooks to each attribute which will execute after that is initialized etc. That should be a separate PR IMO . I'll be glad to add that in a new PR. This PR just adds the attrs, not It's usecases :) .